lib/supavisor/client_handler.ex

```diff
@@ -623,7 +629,8 @@ defmodule Supavisor.ClientHandler do
   end
 
   # pool's manager went down
-  def handle_event(:info, {:DOWN, _, _, _, reason}, state, data) do
+  def handle_event(:info, {:DOWN, ref, _, _, reason}, state, %{manager: manager_ref} = data)
+      when ref == manager_ref do
```

Comment:
I am fan of

```suggestion
  def handle_event(:info, {:DOWN, ref, _, _, reason}, state, %{manager: ref} = data) do
```

But I know that some people do not agree with me there.